### PR TITLE
Update Readme

### PR DESCRIPTION
**Pull request by drisspg**
Stacked PRs:
 * __->__#1526

--- --- ---

Update Readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,19 +19,19 @@ torchao just works with `torch.compile()` and `FSDP2` over most PyTorch models o
...
 2. Quantize the weights and activations: works best for compute bound models
-2. Quantize the activations and weights and sparsify the weight
+3. Quantize the activations and weights and sparsify the weight
 
 ```python
-from torchao.quantization.quant_api import (
+from torchao.quantization import (
     quantize_,
     int8_dynamic_activation_int8_weight,
+    float8_dynamic_activation_float8_weight
```
Comment (jerryzh168) [inline README.md:33]:
```suggestion
    float8_dynamic_activation_float8_weight,
```

File: README.md [hunk 2]
```diff
@@ -128,68 +131,64 @@ optim = CPUOffloadOptimizer(model.parameters(), torch.optim.AdamW, fused=True)
...
+Other options:
+```bash
+# Nightly build
+pip install --pre torchao --index-url https://download.pytorch.org/whl/nightly/cu124
 
-1. [fp6](torchao/dtypes/floatx) for 2x faster inference over fp16 with an easy to use API `quantize_(model, fpx_weight_only(3, 2))`
-2. [2:4 Sparse Marlin GEMM](https://github.com/pytorch/ao/pull/733) 2x speedups for FP16xINT4 kernels even at batch sizes up to 256
-3. [int4 tinygemm unpacker](https://github.com/pytorch/ao/pull/415) which makes it easier to switch quantized backends for inference
+# Different CUDA versions
+pip install torchao --index-url https://download.pytorch.org/whl/cu121  # CUDA 12.1
```
Comment (jerryzh168) [inline README.md:149]: we are supporting 12.4 by default now I think, we also moved 12.1 in CI at some point

File: README.md [hunk 3]
```diff
@@ -128,68 +131,64 @@ optim = CPUOffloadOptimizer(model.parameters(), torch.optim.AdamW, fused=True)
...
-If you believe there's other CUDA kernels we should be taking a closer look at please leave a comment on [this issue](https://github.com/pytorch/ao/issues/697)
+```
 
+### Development Install
+```
+USE_CPP=0 python setup.py develop  # Skip C++/CUDA extensions
+```
 
-## Alpha features
+## Composability
```
Comment (supriyar) [inline README.md:160]: should this also mention DTensor composability?

File: README.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 [![](https://dcbadge.vercel.app/api/server/gpumode?style=flat)](https://discord.gg/gpumode)
 
-[Introduction](#introduction) | [Inference](#inference) | [Training](#training)  | [Composability](#composability) | [Custom Kernels](#custom-kernels) | [Alpha Features](#alpha-features) | [Installation](#installation) | [Integrations](#integrations) | [Videos](#videos) | [License](#license) | [Citation](#citation)
+[Introduction](#introduction) | [Inference](#inference) | [Training](#training)  | [Installation](#installation) |[Composability](#composability) | [Custom Kernels](#custom-kernels) | [Prototype Features](#prototype-features)  | [Integrations](#integrations) | [Videos](#videos) | [License](#license) | [Citation](#citation)
```
Comment (supriyar) [inline README.md:5]: should we highlight some benchmark numbers in the main readme like for int4 and float8 inference and composing sparsity + quantization, currently they are all one-click away?

Comment (drisspg) [reply README.md:5]: added a blurb

File: README.md (hunk 3 above)
Comment (jerryzh168) [reply README.md:167]: yeah I think so, @drisspg you can check out the sglang blog post for some desciprtions

File: README.md [hunk 5]
```diff
@@ -52,13 +60,14 @@ We also provide a developer facing API so you can implement your own quantizatio
...
+Post-training quantization can result in a fast and compact model, but may also lead to accuracy degradation. We recommend exploring Quantization Aware Training (QAT) to overcome this limitation. In collaboration with [Torchtune](https://github.com/pytorch/torchtune/blob/main/recipes/quantization.md#quantization-aware-training-qat), we've developed a QAT recipe that demonstrates significant accuracy improvements over traditional PTQ, recovering **96% of the accuracy degradation on hellaswag and 68% of the perplexity degradation on wikitext** for Llama3 compared to post-training quantization (PTQ). And we've provided a full recipe [here](https://pytorch.org/blog/quantization-aware-training/)
```
Comment (andrewor14) [inline README.md:70]: should we keep the link to the QAT README?

File: README.md [hunk 6]
```diff
@@ -128,68 +139,65 @@ optim = CPUOffloadOptimizer(model.parameters(), torch.optim.AdamW, fused=True)
...
 
-## Alpha features
+[FSDP2](https://github.com/pytorch/torchtitan/blob/main/docs/fsdp.md): Historically most quantization has been done for inference, there is now a thriving area of research combining distributed algorithms and quantization.
 
-Things we're excited about but need more time to cook in the oven
+The best example we have combining the composability of lower bit dtype with compile and fsdp is [NF4](torchao/dtypes/nf4tensor.py) which we used to implement the [QLoRA](https://www.youtube.com/watch?v=UvRl4ansfCg) algorithm. So if you're doing research at the intersection of this area we'd love to hear from you.
 
-1. [MX](torchao/prototype/mx_formats) training and inference support with tensors using the [OCP MX spec](https://www.opencompute.org/documents/ocp-microscaling-formats-mx-v1-0-spec-final-pdf) data types, which can be described as groupwise scaled float8/float6/float4/int8, with the scales being constrained to powers of two. This work is prototype as the hardware support is not available yet.
```
Comment (vkuzo) [inline README.md:153]: it would be nice to keep mx and int8 quantized training callouts as we do plan to bring them out of prototype

Comment (drisspg) [reply README.md:153]: Lets have a prototype Readme and once they are moved out of prototype lets update the readme?

File: README.md (hunk 5 above)
Comment (drisspg) [reply README.md:70]: Hey sorry this was likely rebase bug

File: README.md (hunk 6 above)
Comment (vkuzo) [reply README.md:153]: I think a user should be able to text search the main README.md and know where to go for both MX training/inference and int8 training.  Can we keep short references in the main readme please, with links to more info?  Moving the more descriptive sections to a prototype readme sounds fine.

> Lets have a prototype Readme and once they are moved out of prototype lets update the readme?

Can this be in this PR, instead of deleting?

Comment (vkuzo) [reply README.md:153]: if we just have someting like 

Prototype folder contains upcoming features such as MX training and inference (link) and int8 quantized training (link), and more.

that would sgtm.

**Comment by bhack** [comment]
Can we expand also few lines on autoquant that currently it is really a single line in the `Readme`?

E.g. I see some basic option here that it would nice to clarify as a basic use case:
https://github.com/pytorch/ao/blob/3eb18e771bc7e830a2e56002407256052d8c5e7d/examples/sam2_amg_server/server.py#L389-L418

File: README.md [hunk 7]
```diff
@@ -1,93 +1,82 @@
...
 * 9.5x inference speedups for Image segmentation models with [sam-fast](https://pytorch.org/blog/accelerating-generative-ai)
 * 10x inference speedups for Language models with [gpt-fast](https://pytorch.org/blog/accelerating-generative-ai-2)
 * 3x inference speedup for Diffusion models with [sd-fast](https://pytorch.org/blog/accelerating-generative-ai-3)
 
-torchao works for training too, with [up to 1.5x e2e speedups](https://pytorch.org/blog/training-using-float8-fsdp2/) on large scale (512 GPU / 405B parameter count) pretraining jobs with `torchao.float8`!
-
-torchao just works with `torch.compile()` and `FSDP2` over most PyTorch models on Huggingface out of the box.
-
-## Inference
+`torchao` isn't just for inference, with [up to 1.5x e2e speedups](https://pytorch.org/blog/training-using-float8-fsdp2/) on large scale (512 GPU / 405B parameter count) pretraining jobs with `torchao.float8`!
```
Comment (supriyar) [inline README.md:17]: should we instead reference this blog https://pytorch.org/blog/accelerating-large-scale-training-and-convergence-with-pytorch-float8-rowwise-on-crusoe-2k-h200s/ with support for 2K GPUs and corresponding speedup?

File: README.md [hunk 8]
```diff
@@ -1,93 +1,82 @@
...
+## Inference
 
-You can find a more comprehensive usage instructions for quantization [here](torchao/quantization/) and for sparsity [here](/torchao/_models/sam/README.md).
+`torchao` delivers substantial performance gains with minimal code changes:
 
-#### Quantizing for inference with huggingface configs
+### Performance Highlights
 
-See [docs](https://huggingface.co/docs/transformers/main/en/quantization/torchao) for more details.
+- **INT4 Weight-Only Quantization**: 2x throughput (201 vs 107 tokens/sec) with 65% less memory (4.9GB vs 13.9GB) on LLaMA-2-7B
```
Comment (supriyar) [inline README.md:28]: do we have int4 numbers on llama3 as well?

File: README.md [hunk 9]
```diff
@@ -1,93 +1,82 @@
...
 
-1. Quantize your model with a [post training quantization strategy](#post-training-quantization).
-2. Save your model to disk or upload to huggingface hub ([instructions]( https://huggingface.co/docs/transformers/main/en/quantization/torchao?torchao=manual#serialization)).
-3. [Install](https://github.com/EleutherAI/lm-evaluation-harness?tab=readme-ov-file#install) lm-eval.
-4. Run an evaluation. Example:
+Deploy quantized models with one command:
 
-```bash
-lm_eval --model hf --model_args pretrained=${HF_USER}/${MODEL_ID} --tasks hellaswag --device cuda:0 --batch_size 8
+```shell
```
Comment (supriyar) [inline README.md:67]: feels a bit strange to have the HF command for llama3 and this one for Phi-4

File: README.md [hunk 10]
```diff
@@ -142,89 +131,88 @@ We've added support for semi-structured 2:4 sparsity with **6% end-to-end speedu
...
 ## OSS Integrations
 
 We're also fortunate to be integrated into some of the leading open-source libraries including
 1. Hugging Face transformers with a [builtin inference backend](https://huggingface.co/docs/transformers/main/quantization/torchao) and [low bit optimizers](https://github.com/huggingface/transformers/pull/31865)
-2. Hugging Face diffusers best practices with torch.compile and torchao in a standalone repo [diffusers-torchao](https://github.com/sayakpaul/diffusers-torchao)
+2. Hugging Face diffusers best practices with torch.compile and torchao in a standalone repo [diffusers-torchao](https://github.com/huggingface/diffusers/blob/main/docs/source/en/quantization/torchao.md)
 3. Mobius HQQ backend leveraged our int4 kernels to get [195 tok/s on a 4090](https://github.com/mobiusml/hqq#faster-inference)
-4. [TorchTune](https://github.com/pytorch/torchtune) for our QLoRA and QAT recipes
+4. [TorchTune](https://pytorch.org/torchtune/main/tutorials/qlora_finetune.html?highlight=qlora) for our QLoRA and QAT recipes
 5. [torchchat](https://github.com/pytorch/torchchat) for post training quantization
```
Comment (supriyar) [inline README.md:214]: let's remove torchchat, given that the repo is no longer in development

**Review by supriyar — APPROVED** [review]
left a few comments, please review. Overall looks much better!

File: README.md (hunk 9 above)
Comment (drisspg) [reply README.md:67]: agreed,  will make it consistent
